### ISSUE TITLE
"login is invalid" while using the github.com to authorize

### ISSUE THREAD

**Issue by Stark-X** [kaiyuanshe/kcoin#87]
When I click the button "使用Github账户登录"，the URL changed as below, but it returns a message "login is invalid"
```
http://kcoin.kaiyuanshe.cn/auth/github/callback?redirect_uri=/project&code=<token>&state=<token>
```

**Comment by Stark-X** [comment]
I've started the web project, and the issue could be reproduce, the logs shows as below.
```
127.0.0.1 - - [07/Dec/2018:21:50:33 +0800] "GET /auth/github/login?redirect_uri=%2Fproject HTTP/1.1" 302 - 0.0015
binding user. 0 UserEmail found: []
login is invalid
127.0.0.1 - - [07/Dec/2018:21:50:54 +0800] "GET /auth/github/callback?redirect_uri=/project&code=<code_hash>&state=<state_hash> HTTP/1.1" 409 16 16.6591
```
I am checking the detail for the issue.

**Comment by Stark-X** [comment]
Ahh.. I've checked and debugged last night, then I found the exception is raised from the "src/web/helpers/user_helpers.rb", line 182 ("oauth.update(user_id: user.id)").

 The message *"login is invalid"* means that while the value of the `user.login` is used to update the `OAuth`, that it cannot pass the `OAuth` model's validation.

The validation is using a regex pattern "Username" from "src/web/lib/regex_pattern.rb", the dash in my login name does not match that pattern, so that I can't login via the OAuth function.

The solution is updating the regex pattern to allow the dash in the login name. 
**And the #93 should fix this issue.**

BTW, maybe we should redirect the user to a friendly warning page when the OAuth authorization failed.
